### ISSUE TITLE
Expand supported Losses for first order Extensions

### ISSUE THREAD

**Issue by frhrdr** [f-dangel/backpack#32]
Since first-order extensions allow the use of most nonparametric operations, the number of supported loss functions shouldn't be so small. If the package contained a loss wrapper of the form
L(x) = mean(x),
where x is a vector of the per-sample losses, this would extend the number of available loss functions a lot. These could then be computed as L(f(x)), where f is the chosen loss.
Note, that this can already be done using the MSE loss through MSE(sqrt(f(x)), 0) if f(x) is a vector, but this is not very clean and involves unnecessary computation, so a designated loss function for this purpose would be nice.

**Comment by fKunstner** [comment]
Hi Fred, thanks for the interest! 

If I got your point, it already works and we need to improve the doc. 

For first order extensions, say `BatchGrad`, only the parametric modules need to be `extend`ed. Everything else should work out of the box, see example below.

If you find something that does not work like that, we'd love to know (and fix it). If I completely missed the point, a code example would be useful.

Cheers!

---

For example, an "arbitrary" loss function on a linear model on MNIST data
```
from torch.nn import Flatten, Linear, Sequential
from backpack import backpack, extend, extensions

X = torch.randn(size=(50, 784), requires_grad=True)
model = Sequential(Flatten(), extend(Linear(784, 10)),)
loss = torch.mean(torch.sqrt(torch.abs(model(X))))

with backpack(extensions.BatchGrad()):
    loss.backward()

for name, param in model.named_parameters():
    print(param.grad_batch.shape)
```

**Comment by frhrdr** [comment]
Oh, I see! From the doc I got the impression that I'd have to `extend` the loss function in order to make this work. In that case, adding a short explanation of this in the doc would be enough of course!
